### ISSUE TITLE
MAKE_C_FLAGS should probably have a default

### ISSUE THREAD

**Issue by marcoscaceres** [htacg/tidy-html5#460]
As the project uses c89, I'm wondering if the following should be the default to:

```
-DCMAKE_C_FLAGS:STRING="-std=c89 -pedantic"
```

**Comment by benkasminbullock** [comment]
Those compiler options are specific to GCC (and Clang), so could cause errors when using Microsoft compilers.

**Comment by marcoscaceres** [comment]
Is there equivalent for windows?

**Comment by geoffmcl** [comment]
@marcoscaceres I agree 100% with this/these options for gcc/clang compilers... that support them...

The current CMakeLists.txt has two places where these options can be added... 

````
if(CMAKE_COMPILER_IS_GNUCXX)
    set( WARNING_FLAGS -Wall )
endif(CMAKE_COMPILER_IS_GNUCXX)

if (CMAKE_CXX_COMPILER_ID STREQUAL "Clang") 
   set( WARNING_FLAGS "-Wall -Wno-overloaded-virtual" )
endif() 
````

Or a third, ...
```
if(WIN32 AND MSVC)
    # note dual, in Windows, and using MSVC...
else()
    # add any gcc flags
endif()
```

Really need help in setting the correct `gcc` flags,
as @arnaud always had in **his** build scripts... is it just the 2
flags - `-std=c89 -pedantic` - to be added? 

Welcome a patch, PR, ... on this gcc issue...

Concerning Windows/MSVC there are no `-std=...` flags, that
I know of... as the msvc advanced it supported, more to c11 
than before... seems msvc140 fully supports `c11`?!?!

But I remember, from somewhere, there may be a MSVC flag 
to treat the file as `pure` 'C'... at least as I remember,
in the early MSVC builds, it too would flag a warning about 
at least the use of `\\` inline comments...

Some help in finding this `flag` again would be appreciated...

So, yes, I agree `CMAKE_C_FLAGS` should probably have a default,
please help us in getting there... thanks...

**Comment by marcoscaceres** [comment]
So, naive question: if c11 is backwards compatible with c89, and c99, and msvc140 fully supports c11, why not just target c11? Are there older systems that we need to support that can't handle c11 as a compile target?

**Comment by geoffmcl** [comment]
@marcoscaceres sorry for the delay, but it took me some time to get around to more `testing`, particularly in linux... used Ubuntu 14.04, gcc 4.8.4, Ubuntu 16.04, gcc 5.4.0, and RPI2, Raspbian 8.0 jessie, gcc 4.9.2...

Presently tidy `master` 5.3.12 **compiles** cleanly in the above systems, as it does in windows...

## Windows and MSVC versions

In MSVC, as I understand it, some c99/c11 changes/features were supported quite a while back. Some more were added in MSVC14 (2015), but is it fully `c99/c11` compliant? I doubt it, but have not hit any problems yet...  

The last time I checked, back about 4 months, or so, tidy would compile with MSVC10, aka _MSC_VER 1600, 2010... I now do not have any earlier version available, but tidy did used to compile in MSVC6 (2005)...

So this does not seem to be a windows problem... I would be very happy if we maintained compatibility back to say MSVC10, which it seems, is what we have now...

## unix/linux

But considering the `gcc` compiler, it seem this can sort of be divided into two(2) issues -

 1. Use of `//` inline comments, arrays, string length, ... -pedantic
 2. Adding specific standard -std=c89, c99, c11, ...

#### -pedantic

If I add -pedantic, and fix the one very noisy warning in include/tidy.h - `// bit like OPTIONS` - there are not too many other `//` uses, and these can be easily fixed... and I think should be done... but am open on this `comment` issue...

There is the quite disturbing warning set from src/language.c, which has quite a number of strings greater than `509`, and the use of a flexible array... Do these need to be **fixed**? How do you support strings of greater than `509`, which tidy needs to, to support different language sets?

So, in principal have no problem that `-pedantic` being added to the CMakeLists.txt, making it the default, but we would have to **fix** `language.c` first. 

What do others think?

#### standards -std=c89, c99, c11, ...

It seems if I add any of these `-std=xXX` options, the compile FAILS! The noisiest error is missing definition of `uint`!

Tidy has been using a `uint` since the beginning of time... it is present in `tidy4aug00` code, now 16 years ago. What! Now, 16 years later, gcc decides to not support this? What is this all about?

It does support it if you do not add this `-std=cNN` option! This, I do not understand?

Yes, as I remember from an earlier issue, we could do a massive search and replace every `uint` with something else, but what? Or what header has to now be added to the build to get this defined? Why is this happening?

Can someone **help** with exactly what has to be done here?

**Comment by geoffmcl** [comment]
@marcoscaceres, oops, maybe I overlooked what is in your PR #463... sorry...

I will try to find the time to pull your patches... test, and look... maybe you have answered some of my questions above... thanks...

**Comment by geoffmcl** [comment]
As we have now merged PR #463 which removed all the single line comment, `//`, assume we can close this...

Please feel free to re-open, or file a new issue... thanks...
